os/board/rtl8730e/src/rtl8730e_st7785_st7701.c

```diff
@@ -0,0 +1,293 @@
+	ResetPin.GPIO_PuPd = GPIO_PuPd_NOPULL;
+	ResetPin.GPIO_Mode = GPIO_Mode_OUT;
+	GPIO_Init(&ResetPin);
+#ifdef CONFIG_LCD_ST7785
+	pwmout_init(&g_rtl8730_config_dev_s.pwm_led, GPIO_PIN_BACKLIGHT);
+	lcddbg("initial pwm read %f\n", pwmout_read(&g_rtl8730_config_dev_s.pwm_led));
+#endif
+}
+
+static void rtl8730_control_backlight(uint8_t level)
+{
+
+	float pwm_level = level/100.0;
+	lcddbg("level :%d , pwm level:%f\n", level, pwm_level);
+#ifdef CONFIG_LCD_ST7785
+	pwmout_write(&g_rtl8730_config_dev_s.pwm_led, 1.0-pwm_level);
+#endif
+}
+
+static void rtl8730_enable_lcdc(void)
+{
+	LCDC_Cmd(pLCDC, ENABLE);
+	while (!LCDC_CheckLCDCReady(pLCDC)) ;
+	mipidsi_mode_switch(true);
+}
+
+void rtl8730_mipidsi_underflowreset(void)
+{
+	u32 reg_val2 = MIPI_DSI_INTS_ACPU_Get((MIPI_TypeDef *) mipi_irq_info.data);
+
+	if (reg_val2) {
+		UnderFlowCnt = 0;
+		MIPI_DSI_INT_Config((MIPI_TypeDef *) mipi_irq_info.data, DISABLE, DISABLE, DISABLE);
+		mipidsi_acpu_reg_clear();
+
+		/*Disable the LCDC*/
+		LCDC_Cmd(pLCDC, DISABLE);
+		lldbg("ERROR: LCDC_CTRL 0x%x\n", pLCDC->LCDC_CTRL);
+
+		rtl8730_enable_lcdc();
+	}
+}
+
+u32 rtl8730_hv_isr(void *Data)
+{
+	/*get LCDC interrupt status*/
+	volatile u32 IntId = LCDC_GetINTStatus(pLCDC);
+	/*select operations according to interrupt ID*/
+	if (IntId & LCDC_BIT_LCD_LIN_INTS) {
+		//lldbg("line interrupt occur\n");
+		LCDC_ClearINT(pLCDC, LCDC_BIT_LCD_LIN_INTS);
+	}
+
+	if (IntId & LCDC_BIT_LCD_FRD_INTS) {
+                //lldbg("refresh interrupt occur\n");
+                LCDC_ClearINT(pLCDC, LCDC_BIT_LCD_FRD_INTS);
+        }
+
+	if (IntId & LCDC_BIT_DMA_UN_INTS) {
+		//lldbg("DMA interrupt occur\n");
+		LCDC_ClearINT(pLCDC, LCDC_BIT_DMA_UN_INTS);
+		UnderFlowCnt++;
+		if (UnderFlowCnt == 1) {
+			lcddbg("ERROR: LCDC DMA Underflow-----\n");
+			InterruptRegister((IRQ_FUN)rtl8730_mipidsi_underflowreset, mipi_irq_info.num, (u32)mipi_irq_info.data, mipi_irq_info.priority);
+			InterruptEn(mipi_irq_info.num, mipi_irq_info.priority);
+			mipidsi_acpu_reg_clear();
+			mipidsi_mode_switch(false);
+			MIPI_DSI_INT_Config(MIPI, ENABLE, ENABLE, ENABLE);
+		}
+	}
+	return 0;
+}
+
+static void rtl8730_register_lcdc_isr(void){
+	InterruptDis(lcdc_irq_info.num);
+	InterruptUnRegister(lcdc_irq_info.num);
+	InterruptRegister((IRQ_FUN)rtl8730_hv_isr, lcdc_irq_info.num, (u32)lcdc_irq_info.data, lcdc_irq_info.priority);
+	InterruptEn(lcdc_irq_info.num, lcdc_irq_info.priority);
+}
+
+void rtl8730_lcdc_initialize(void)
+{
+	rtl8730_gpio_init();
+	struct lcd_data config;
+	config.XPixels = XRES;
+	config.YPixels = YRES;
+	config.mipi_frame_rate = MIPI_FRAME_RATE;
+	config.mipi_dsi_HBP = MIPI_DSI_HBP;
+	config.mipi_dsi_HFP = MIPI_DSI_HFP;
+	config.mipi_dsi_HSA = MIPI_DSI_HSA;
+	config.mipi_dsi_RTNI = MIPI_DSI_RTNI;
+	config.mipi_dsi_VBP = MIPI_DSI_VBP;
+	config.mipi_dsi_VFP = MIPI_DSI_VFP;
+	config.mipi_dsi_VSA = MIPI_DSI_VSA;
+	config.lcd_lane_num = MIPI_LANE_NUMBER;
+	rtl8730_gpio_reset();
+	rtl8730_control_backlight(100);
```

Comment:
Behavior include backlight control should be handled by above layer.
hence st7785/st7701_lcdinitialize should control it. (perhaps config.backlight(100)?)